### DISCUSSION TITLE
Very long loading time when defining people (face recognition) - please help!

### DISCUSSION THREAD

**Discussion by hpoperator86** [photoprism/photoprism#4173] [Q&A]
Hello dear community,

I currently have a problem with Photoprism.
I have successfully set up Photoprism under Docker (with a docker-compose, which I include here) and successfully indexed (not imported) my images. The total number of indexed content is 26167, which is quite a lot.
The Photoprism docker runs on a Fujitsu PRIMERGY TX 1310 M1 (Intel Xeon CPU E3-1226 v3 @ 3.30GHz , 8GB DDR3L 1600MHz ECC (HMT41GU7BFR8A-PB) ). So no Raspberry or similarly weak. The main instance is running Openmedivault-OS.

The face recognition or the subsequent definition under "People" --> "+New" --> "Show all new faces" takes an extremely long time (it takes exactly 1 minute and 32 seconds). I press on the respective photo there, then on the pencil symbol and there under People I mark the previously saved person. It then takes a super long time to load. 
When I consider that I have to assign all the people bit by bit, I sit at photoprism for days or weeks without interruption.

During recognition, the CPU load of my host increases to approx. 33% and then to 0.8%; RAM utilization is a steady 3.49 GiB (46%).
So in my opinion everything is fine. 
I just don't understand the bad performance. Please help!

attached my docker-compose:

`services:
  photoprism:
    image: photoprism/photoprism:latest
    restart: unless-stopped
    stop_grace_period: 10s
    depends_on:
      - mariadb
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    ports:
      - "2342:2342"
    env_file:
      - stack.env
    working_dir: "/photoprism"
    volumes:
      - "/srv/dev-disk-by-uuid-455ecddd-d762-4c16-8b87-4f47721f9208/Hauptordner1/Bilder:/photoprism/originals" # Originals mount path can be changed as needed (DO NOT REMOVE)
      - "/srv/dev-disk-by-uuid-2e5aebd3-e509-4523-8976-d8095258f48a/Hauptordner2/Backups_smartphone/Andy/Samsung Galaxy S21/DCIM:/photoprism/originals/Samsung_Galaxy_S21_Backup"
      - "/srv/dev-disk-by-uuid-2e5aebd3-e509-4523-8976-d8095258f48a/Hauptordner2/Backups_smartphone/Soravi/Mi10/DCIM:/photoprism/originals/Mi10_Backup"
      - "/srv/dev-disk-by-uuid-1157d268-3074-4683-80d4-dad13fe963cf/Hauptordner3/photoprism/storage:/photoprism/storage" 
   

  mariadb:
    image: mariadb:11
    restart: unless-stopped
    stop_grace_period: 5s
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    command: --innodb-buffer-pool-size=512M --transaction-isolation=READ-COMMITTED --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci --max-connections=512 --innodb-rollback-on-timeout=OFF --innodb-lock-wait-timeout=120
     volumes:
      - "/srv/dev-disk-by-uuid-1157d268-3074-4683-80d4-dad13fe963cf/Hauptordner3/photoprism/appdata:/var/lib/mysql"
    env_file:
      - stack.env`

also attached my settings of the env as a screenshot:
![Screenshot 2024-04-13 115503](https://github.com/photoprism/photoprism/assets/149537735/bb84576a-4304-457b-a437-697b2545b036)
![Screenshot 2024-04-13 115521](https://github.com/photoprism/photoprism/assets/149537735/801eedd9-02c9-446b-ad74-76ee77365a18)
![Screenshot 2024-04-13 115535](https://github.com/photoprism/photoprism/assets/149537735/8b961bdc-8c08-466f-9e91-6a9ce4dff056)

**Comment by graciousgrey** [comment]
Hello :wave: 

We will take care of this issue as soon as we have released the new authentication options we are currently working on.

- https://github.com/photoprism/photoprism/issues/3124

**Comment by jbbandos** [comment]
I'm also running photoprism on OMV, but in my case using the OMV plugin that uses podman instead of docker. 
I did get a major improvement in speed by moving the database, as well as the configs, to a filesystem on SATA SSD instead of using the "normal" HDD storage. I have over 80000 photos on my library right now (even though ~30000 are duplicates). Now, if I could understand how to pass parameters to the database in this setup I'm guessing I'd be able to tune some more speed out of it.
